### PR TITLE
Update tahoma and mfc42 verb: use fileExists

### PR DESCRIPTION
**Pull request by ImperatorS79**
### Description
Fix a part of #1107 relating mfc42.

### Ready for review
- [x] Script tested as a regular phoenicis user and working (if you have a problem -> create as draft and ask for help).
- [x] `json-align` and `eslint` run according to the [documentation](https://phoenicisorg.github.io/scripts/General/tools/). 
- [x] Codacy and travis checked.

### CODE REVIEW

**Comment by ImperatorS79** [comment]
@Zemogiter could you test if this fixes your errors with mfc42 and tahoma ?

**Comment by madoar** [comment]
@ImperatorS79 I would like to wait until #1112 has been merged before we merge this

**Comment by Zemogiter** [comment]
@ImperatorS79 mfc42 works but in `Tahoma` we need extra `)` in https://github.com/PhoenicisOrg/scripts/blob/d3ab4e24ddb36f4f41b1d1a0e682a1aca03a190b/Engines/Wine/Verbs/Tahoma/script.js#L27

**Comment by ImperatorS79** [comment]
Could someone approve this ?

**Comment by ImperatorS79** [comment]
@Zemogiter can you test again ?

File: Engines/Wine/Verbs/Tahoma/script.js [hunk 1]
```diff
@@ -19,6 +19,10 @@ class Tahoma {
         const wizard = this.wine.wizard();
         const prefixDirectory = this.wine.prefixDirectory();
         const fontDirectory = this.wine.fontDirectory();
+        
+        if (!fileExists(fontDirectory)) {
```
Comment (madoar) [inline Engines/Wine/Verbs/Tahoma/script.js]: The method name sounds strange in this case. Should we add a dedicated method for folders? What do you think @plata @ImperatorS79?

Comment (plata) [reply Engines/Wine/Verbs/Tahoma/script.js]: We can do that. Would basically be an alias then, right?

**Comment by plata** [comment]
Please fix the PR title before merging.

File: Engines/Wine/Verbs/Tahoma/script.js (hunk 1 above)
Comment (madoar) [reply Engines/Wine/Verbs/Tahoma/script.js]: Actually I'm not sure. If we decide to split the functionality into a `fileExists` and `directoryExists` method we could also test whether the given path is not a file/directory but instead a directory/file, depending on the regarded method.

**Comment by Zemogiter** [comment]
@ImperatorS79 at the begining of .cab file extraction I get the error saying the fonts directory does not exist but it does
```
Extracting to: /home/jonasz/.Phoenicis/containers//wineprefix//RimWorld//drive_c/tahoma/
/home/jonasz/.Phoenicis/resources///IELPKTH.CAB: WARNING; possible 5592 extra bytes at end of file.
Extracting cabinet: /home/jonasz/.Phoenicis/resources///IELPKTH.CAB

All done, no errors.
[ERROR] org.phoenicis.multithreading.ControlledThreadPoolExecutorService (l.64) - Path "/home/jonasz/.Phoenicis/containers//wineprefix//RimWorld//drive_c/windows/Fonts" does not exist
	at org.phoenicis.tools.files.FileUtilities.copy(FileUtilities.java:139)
	at <js> cp(Unnamed:56:1251-1284)
	at <js> go(Unnamed:40:1231-1295)
	at <js> install(Unnamed:57:1761-1781)
	at org.graalvm.sdk/org.graalvm.polyglot.Value.invokeMember(Value.java:457)
	at org.phoenicis.engines.VerbsManager.lambda$installVerb$0(VerbsManager.java:71)
	at org.phoenicis.scripts.session.PhoenicisInteractiveScriptSession.eval(PhoenicisInteractiveScriptSession.java:35)
	at org.phoenicis.scripts.interpreter.BackgroundScriptInterpreter.lambda$createInteractiveSession$1(BackgroundScriptInterpreter.java:45)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by host exception: java.lang.IllegalArgumentException: Path "/home/jonasz/.Phoenicis/containers//wineprefix//RimWorld//drive_c/windows/Fonts" does not exist

Exception in thread "pool-3-thread-5" Path "/home/jonasz/.Phoenicis/containers//wineprefix//RimWorld//drive_c/windows/Fonts" does not exist
	at org.phoenicis.tools.files.FileUtilities.copy(FileUtilities.java:139)
	at <js> cp(Unnamed:56:1251-1284)
	at <js> go(Unnamed:40:1231-1295)
	at <js> install(Unnamed:57:1761-1781)
	at org.graalvm.sdk/org.graalvm.polyglot.Value.invokeMember(Value.java:457)
	at org.phoenicis.engines.VerbsManager.lambda$installVerb$0(VerbsManager.java:71)
	at org.phoenicis.scripts.session.PhoenicisInteractiveScriptSession.eval(PhoenicisInteractiveScriptSession.java:35)
	at org.phoenicis.scripts.interpreter.BackgroundScriptInterpreter.lambda$createInteractiveSession$1(BackgroundScriptInterpreter.java:45)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by host exception: java.lang.IllegalArgumentException: Path "/home/jonasz/.Phoenicis/containers//wineprefix//RimWorld//drive_c/windows/Fonts" does not exist
```

**Comment by ImperatorS79** [comment]
Yes, I have the same kind of error in other verbs when testing Adobe Photoshop script, that says system32 does not exist. There is something we do not understand here...

**Comment by madoar** [comment]
The error only occurs when a `CabExtract` is followed by a `cp` instruction right?
If this is the case then maybe `CabExtract` returns before the filesystem has been updated/refreshed?

Can you try adding a `java.lang.Thread.sleep(100);` instruction between `CabExtract` and `cp`?

**Comment by Zemogiter** [comment]
@madoar done that, still getting the same error

**Comment by madoar** [comment]
I'm still confuses from which line in which script this error is thrown... The issue is that the source path/directory in a `cp` operation does not exist. So which script calls `cp(fontDirectory, ...)` in line 55? I am unable to even find a script that calls `cp(fontDirectory, ...)` at all...

**Comment by plata** [comment]
You can add some log output like
```Javascript
print("1");
// do something
print("2");
// do something more
```
to figure out where exactly the error is caused.

**Comment by madoar** [comment]
Are there any updates on this?

**Comment by madoar** [comment]
Yes the PR needs to be rebased. The changes to the plugins are not included.

**Comment by Zemogiter** [comment]
@ImperatorS79 pls rebase

**Comment by Zemogiter** [comment]
Too bad we can't use dependabot in non-dependabot PRs to rebase them.

**Comment by plata** [comment]
https://github.com/cirrus-actions/rebase

**Comment by Zemogiter** [comment]
@ImperatorS79 please remove trailing spaces

**Comment by plata** [comment]
If you're interested in having this rebase action, please open an issue regarding it. Then we can discuss.

**Pull request merged**